### PR TITLE
Add CopyFunction on response card

### PR DESCRIPTION
**Pull request by Danielecina**
#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `npm test`
- [x] tests are included
- [x] the documentation is updated or integrated accordingly with your changes
- [x] the code will follow the lint rules and style of the project
- [x] you are not committing extraneous files or sensible data

### CODE REVIEW

File: packages/api-explorer/src/components/Response/index.jsx [hunk 1]
```diff
@@ -11,8 +11,24 @@ const ResponseBody = require('./ResponseBody');
 const showCodeResults = require('../../lib/show-code-results');
 
 const Oas = require('../../lib/Oas');
+const CopyCode = require('../CopyCode');
 
 const { Operation } = Oas;
+const ctaContainerStyle = {
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/index.jsx]: why cta?

Comment (Danielecina) [reply packages/api-explorer/src/components/Response/index.jsx]: I used it to indicate the whole "call to action" button inside the container.
Maybe ContainerCtaStyle is better.

File: packages/api-explorer/src/components/Response/ResponseBody.jsx [hunk 2]
```diff
@@ -50,17 +50,14 @@ Unauthorized.defaultProps = {
 };
 
 // eslint-disable-next-line react/prop-types
-function ResponseBody({ result, isOauth, oauth }) {
+function ResponseBody({ result, isOauth, oauth, isCollapse}) {
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/ResponseBody.jsx]: isCollapsed

File: packages/api-explorer/src/components/Response/ResponseBody.jsx [hunk 3]
```diff
@@ -50,17 +50,14 @@ Unauthorized.defaultProps = {
...
   return (
-    <div style={{ display: 'block'}}>
-      {
-        result.status === 401 && <Unauthorized isOauth={isOauth} oauth={oauth} />
+    <div style={{ display: 'block', color: '#fff'}}>
+      {result.status === 401 && <Unauthorized isOauth={isOauth} oauth={oauth} />}
+      {result.responseBody ?
+        <Result result={result} isCollapse={isCollapse} /> :
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/ResponseBody.jsx]: isCollapsed

File: packages/api-explorer/src/components/Response/Result.jsx [hunk 4]
```diff
@@ -9,37 +10,60 @@ const syntaxHighlighter = require('@mia-platform/syntax-highlighter');
...
-              overflow: 'visible',
-              whiteSpace: 'pre-wrap',
-              wordBreak: 'break-all'
-            }}
-          />
-        )
+      {result.isBinary && (
+        <div style={{display: 'flex', flexDirection: 'column', alignItems: 'center'}}>
+          <div style={{padding: 8}}>
+            <svg xmlns="http://www.w3.org/2000/svg" width="41" height="53" viewBox="0 0 41 53">
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/Result.jsx]: Could we create a constant to address this svg? To have an explicit name

File: packages/api-explorer/src/components/Response/Result.jsx [hunk 5]
```diff
@@ -9,37 +10,60 @@ const syntaxHighlighter = require('@mia-platform/syntax-highlighter');
...
+              <FormattedMessage id="code.download" defaultMessage="Download" />
+            </Button>
+          </div>
+        </div>
+      )}
+
+      {!result.isBinary && isJson && (
+        <ReactJson
+          src={result.responseBody}
+          collapsed={isCollapse === undefined ? 1 : isCollapse}
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/Result.jsx]: This expand all nodes if collapsed prop is false? From `react-json-view` docs it is not clear

Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/Result.jsx]: If isCollapse is false, we should set it to 1. So isCollapsed could be  true/false, with a default props

File: packages/api-explorer/src/components/Response/index.jsx [hunk 6]
```diff
@@ -47,30 +85,36 @@ class Response extends React.Component {
...
-    const placeholderStyle = {
-      textAlign: 'center',
-      padding: '40px 0',
-      color: 'rgba(255,255,255,0.7)',
-      fontStyle: 'italic',
-      fontSize: '14px',
-    }
-   
+
+    console.log(result)
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/index.jsx]: Remove console.log

File: packages/api-explorer/src/components/Response/index.jsx [hunk 7]
```diff
@@ -47,30 +85,36 @@ class Response extends React.Component {
...
-                {responseTab === 'metadata' && <ResponseMetadata result={result} />}
+        {result !== null ? (
+          <BlockWithTab
+            items={itemsResult}
+            selected={responseTab}
+            onClick={this.setTab}
+          >
+            {!result.isBinary ? (
+              <div style={ctaContainerStyle}>
+                <CopyCode code={JSON.stringify(result.responseBody)} />
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/index.jsx]: With a json stringify, this responseBody must be a json. So why the check below?

File: packages/api-explorer/src/Doc.jsx [hunk 8]
```diff
@@ -267,7 +267,11 @@ class Doc extends React.Component {
   renderResponse() {
     return (
       <Response
-        result={this.state.result}
+        result={{
+          isBinary: true,
+          responseBody: 'https://upload.wikimedia.org/wikipedia/commons/b/bc/IKB_191.jpg',
+          type: 'image/jpg'
+        }} // this.state.result}
```
Comment (davidebianchi) [inline packages/api-explorer/src/Doc.jsx]: Why `this.state.result`?

File: packages/api-explorer/src/components/Response/Result.jsx (hunk 5 above)
Comment (davidebianchi) [reply packages/api-explorer/src/components/Response/Result.jsx]: ??

File: packages/api-explorer/src/components/Response/Result.jsx [hunk 9]
```diff
@@ -9,52 +10,74 @@ const syntaxHighlighter = require('@mia-platform/syntax-highlighter');
...
+          </div>
+        </pre>
+      )}
     </div>
   );
 }
 
 Result.propTypes = {
+  // eslint-disable-next-line react/require-default-props
+  isCollapse: PropTypes.bool,
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/Response/Result.jsx]: Set default props to false

File: .travis.yml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 
 node_js:
-  - "node"
+  - 12.9.0
```
[the remark this answers is not in the document]
Comment (Danielecina) [reply .travis.yml]: ok

File: packages/api-explorer/src/Doc.jsx (hunk 8 above)
Comment (Danielecina) [reply packages/api-explorer/src/Doc.jsx]: my fault

File: packages/api-explorer/src/components/Response/Result.jsx (hunk 9 above)
Comment (Danielecina) [reply packages/api-explorer/src/components/Response/Result.jsx]: https://github.com/mia-platform/api-explorer/pull/35#discussion_r332044924

Comment (davidebianchi) [reply packages/api-explorer/src/components/Response/Result.jsx]: If you set it to false in default, you can set the collapsed:
```
isCollapse || 1
```

**Pull request merged**